doc/contrib/recording.md

```diff
@@ -0,0 +1,81 @@
+# Session Recording Technical Documentation
+
+After every `kdbSet` action, the changes are calculated using Elektra's powerful changetracking API.
+The result of the calculation is an `ElektraDiff` instance we'll call _part diff_ throughout this document.
+The session recording plugin merges those _part diffs_ together and creates and persists an overall _session diff_.
+
+Conceptually, this is depicted in the following image:
+
+![Session vs Part diff](../images/elektra-record/recording-part-diff.svg)
+
+Importantly, the _session diff_ is shared across processes.
+The _session diff_ persistently records all changes made to the whole KDB from any process between the start and end of the session.
+Because of the cross-process nature of a _session diff_, it is also susceptible to "simultaneous write" conflicts.
+We tried to limit the impact of these conflicts through locking.
```

Comment:
Sounds like the chosen approach does not work reliably.

Actually we already have lock plugins. Probably they should be used here, too.

Reply:
I agree that this sentence is too pessimistic because our approach actually works perfectly and we aren't producing any conflicts.

But I'm curious: could you point me to those lock plugins? I just took a quick look and didn't find any.

Reply:
Well, perfectly in the sense that it doesn't produce an inconsistent state.
We fail in the exact same way that concurrent writes to the configuration files will fail. Applications already need to handle that. From the application's POV, it doesn't even know whether it failed because of recording or the configuration itself.